src/vm_cpu.rs

```diff
@@ -358,6 +357,37 @@ impl VmPmuState {
             }
         }
     }
+
+    /// Returns the cached value for a PMU CSR
+    pub fn get_cached_csr_value(&self, csr: u64) -> Option<u64> {
+        use drivers::pmu::PmuInfo;
+        use PmuCounterState::Configured;
+        let pmu_info = PmuInfo::get();
+        if let Ok(counter_index) = pmu_info.get_counter_for_csr(csr) &&
+           let Configured(c) = self.counter_state[counter_index as usize] {
+            Some(c.value)
+        } else {
+            None
+        }
```

Comment:
Instead of multiple `let` chains, how about:

```
let counter_index = pmu_info.get_counter_for_csr(csr).ok()? as usize;
if let Configured(c) = self.counter_state[counter_index] {
    Some(c.value)
} else {
    None
}
```

Reply:
Yes indeed...I need to ingrain patterns using more of the ? operator.